### ISSUE TITLE
react-tools jsx ignores -x when src and build dirs are the same

### ISSUE THREAD

**Issue by hatched** [facebook/react#3551]
In an existing project with a complex directory structure of js files we are moving them one at a time over to React components. The existing build system takes all of the js files in this structure and produces the final built files. In these directories we take one js file at a time and convert it to a jsx file.

When a *.jsx file changes, the compiler builds the resulting js file but then re-builds the same file again when the js file is added to the tree.

To reproduce assume the directory structure of:
```
jsx-test
  \src
    source.jsx
```
From within `jsx-test` run `jsx -wx jsx src src` it will build the source.jsx file.
Make a change to the source.jsx file and you'll notice that it builds it twice, the second time after the js file gets added to the tree.

**Comment by zpao** [comment]
We're going to deprecate react-tools so closing this out.

**Comment by hatched** [comment]
Alright, out of curiosity, what will be replacing it? I see the docs still mention react-tools.

**Comment by cody** [comment]
@hatched The react-tools will be replaced by Babel: https://babeljs.io 
Babel is pretty popular, so a lot of people have already switched. It compiles JSX by default.

**Comment by hatched** [comment]
Very cool, thanks!

**Comment by zpao** [comment]
Yea, I haven't blogged yet. react-tools was always mostly just a pretty command line interface to jstransform, so we moved the wrapping bits into jstransform and shipped that. That lets us decouple releases. But then we also decided to stop using jstransform externally so we'll be recommending people start using babel. However, jstransform does exist with mostly the same API as a less churny alternative. https://github.com/facebook/jstransform#migration-guide
